### DISCUSSION TITLE
How to transfer a string into a JSON?

### DISCUSSION THREAD

**Discussion by nickchengster** [nlohmann/json#3192] [Q&A] [answered]
Hi,
In the code, I would like to transfer a received std::string with a format like xxx:yyy,iii:jjj into a JSON.
This string is received from clients' side. 
I got the error messae as [json.exception.parse_error.101] parse error at line 1, column 2: syntax error while parsing value - invalid literal; last read: 'x'.
` 
   std::string className;
`
`
   std::string instanceId;
`
`
   parseUrl(requestUrl, className, instanceId);
`
`
   try {
`
`
      nlohmann::json j = nlohmann::json::parse(instanceId);
`
`
      syslog(LOG_ERR, "instanceId:: xxx: %s", j["xxx"].get<std::string>().c_str());
`
`
   } catch (nlohmann::detail::exception& e) {
`
`
      syslog(LOG_ERR, "message: %s", e.what());
`
`
   }
`
How could I transfer the string into a so-called raw string or something-else which is acceptable by nlohmann::json::parse().

**Comment by nlohmann** [comment, marked as answer]
The parse error indicates that the input is invalid JSON. What is the value of `instanceId` when you call `json::parse`?

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Reply by nickchengster** [reply]
No. Thanks
